sql-plugin/src/main/scala/org/apache/spark/sql/rapids/AggregateFunctions.scala

```diff
@@ -255,6 +255,50 @@ class CudfMin(ref: Expression) extends CudfAggregate(ref) {
   override def toString(): String = "CudfMin"
 }
 
+class CudfCollectList(ref: Expression) extends CudfAggregate(ref) {
+  override lazy val updateReductionAggregate: cudf.ColumnVector => cudf.Scalar =
+    throw new UnsupportedOperationException("CollectList is not yet supported in reduction")
+  override lazy val mergeReductionAggregate: cudf.ColumnVector => cudf.Scalar =
+    throw new UnsupportedOperationException("CollectList is not yet supported in reduction")
+  override lazy val updateAggregate: Aggregation = Aggregation.collectList()
+  override lazy val mergeAggregate: Aggregation = Aggregation.mergeLists()
+  override def toString(): String = "CudfCollectList"
+  override def dataType: DataType = ArrayType(ref.dataType, containsNull = false)
+  override def nullable: Boolean = false
+}
+
+class CudfMergeLists(ref: Expression) extends CudfAggregate(ref) {
+  override lazy val updateReductionAggregate: cudf.ColumnVector => cudf.Scalar =
+    throw new UnsupportedOperationException("MergeLists is not yet supported in reduction")
+  override lazy val mergeReductionAggregate: cudf.ColumnVector => cudf.Scalar =
+    throw new UnsupportedOperationException("MergeLists is not yet supported in reduction")
+  override lazy val updateAggregate: Aggregation = Aggregation.mergeLists()
+  override lazy val mergeAggregate: Aggregation = Aggregation.mergeLists()
+  override def toString(): String = "CudfMergeLists"
+}
+
+class CudfCollectSet(ref: Expression) extends CudfAggregate(ref) {
+  override lazy val updateReductionAggregate: cudf.ColumnVector => cudf.Scalar =
+    throw new UnsupportedOperationException("CollectSet is not yet supported in reduction")
+  override lazy val mergeReductionAggregate: cudf.ColumnVector => cudf.Scalar =
+    throw new UnsupportedOperationException("CollectSet is not yet supported in reduction")
+  override lazy val updateAggregate: Aggregation = Aggregation.collectSet()
```

Comment:
Humnn, is there any way to call `Aggregation.collectList` if the data are partitioned into more than one batch? As we only need collectSet for the intermediate results. Calling collectSet for them is expensive, as that involves unnecessarily executing `drop_list_duplicates` on the intermediate lists.